src/thermo/PengRobinson.cpp

```diff
@@ -0,0 +1,817 @@
+
+    // Update individual alpha
+    for (size_t j = 0; j < m_kk; j++) {
+        double critTemp_j = speciesCritTemperature(m_a_coeffs(j,j), m_b_coeffs[j]);
+        double sqt_alpha = 1 + m_kappa[j] * (1 - sqrt(temp / critTemp_j));
+        m_alpha[j] = sqt_alpha*sqt_alpha;
+    }
+
+    //Update aAlpha_i, j
+    for (size_t i = 0; i < m_kk; i++) {
+        for (size_t j = 0; j < m_kk; j++) {
+            m_aAlpha_binary(i, j) = sqrt(m_alpha[i] * m_alpha[j]) * m_a_coeffs(i,j);
+        }
+    }
+    calculateAB(m_a,m_b,m_aAlpha_mix);
+}
+
+void PengRobinson::calculateAB(double& aCalc, double& bCalc, double& aAlphaCalc) const
+{
+    bCalc = 0.0;
+    aCalc = 0.0;
+    aAlphaCalc = 0.0;
+    for (size_t i = 0; i < m_kk; i++) {
+        bCalc += moleFractions_[i] * m_b_coeffs[i];
+        for (size_t j = 0; j < m_kk; j++) {
+            double a_vec_Curr = m_a_coeffs(i, j);
+            aCalc += a_vec_Curr * moleFractions_[i] * moleFractions_[j];
+            aAlphaCalc += m_aAlpha_binary(i, j) * moleFractions_[i] * moleFractions_[j];
+        }
+    }
+}
+
+double PengRobinson::daAlpha_dT() const
+{
+    double daAlphadT = 0.0, temp, k, Tc, sqtTr, coeff1, coeff2;
+    for (size_t i = 0; i < m_kk; i++) {
+        // Calculate first derivative of alpha for individual species
+        Tc = speciesCritTemperature(m_a_coeffs(i,i), m_b_coeffs[i]);
+        sqtTr = sqrt(temperature() / Tc); //we need species critical temperature
+        coeff1 = 1 / (Tc*sqtTr);
+        coeff2 = sqtTr - 1;
+        k = m_kappa[i];
+        m_dalphadT[i] = coeff1 * (k*k*coeff2 - k);
+    }
+    //Calculate mixture derivative
+    for (size_t i = 0; i < m_kk; i++) {
+        for (size_t j = 0; j < m_kk; j++) {
+            temp = 0.5 * sqrt((m_a_coeffs(i, i) * m_a_coeffs(j, j)) / (m_alpha[i] * m_alpha[j]));
+            daAlphadT += moleFractions_[i] * moleFractions_[j] * temp
+                        * (m_dalphadT[j] * m_alpha[i] + m_dalphadT[i] * m_alpha[j]);
+        }
+    }
+    return daAlphadT;
+}
+
+double PengRobinson::d2aAlpha_dT2() const
+{
+    for (size_t i = 0; i < m_kk; i++) {
+        double Tcrit_i = speciesCritTemperature(m_a_coeffs(i, i), m_b_coeffs[i]);
+        double sqt_Tr = sqrt(temperature() / Tcrit_i); //we need species critical temperature
+        double coeff1 = 1 / (Tcrit_i*sqt_Tr);
+        double coeff2 = sqt_Tr - 1;
+        //  Calculate first and second derivatives of alpha for individual species
+        double k = m_kappa[i];
+        m_dalphadT[i] = coeff1 * (k*k*coeff2 - k);
+        m_d2alphadT2[i] = (k*k + k) * coeff1 / (2*sqt_Tr*sqt_Tr*Tcrit_i);
+    }
+
+    //Calculate mixture derivative
+    double d2aAlphadT2 = 0.0;
+    for (size_t i = 0; i < m_kk; i++) {
+        double alphai = m_alpha[i];
+        for (size_t j = 0; j < m_kk; j++) {
+            double alphaj = m_alpha[j];
+            double alphaij = alphai * alphaj;
+            double temp = 0.5 * sqrt(m_a_coeffs(i, i) * m_a_coeffs(j, j) / (alphaij));
```

Comment:
Here also, this ought to at least be:

```suggestion
            double temp = 0.5 * m_a_coeffs(i, j) / sqrt(alphaij));
```
But I also think there is a slightly simpler implementation, which I'll suggest further below.